include/fiction/algorithms/simulation/sidb/random_layout_generator.hpp

```diff
@@ -0,0 +1,200 @@
+
+#include <iostream>
+#include <vector>
+
+namespace fiction
+{
+/**
+ * This struct stores the parameters for the *generate_random_layouts* algorithm.
+ */
+template <typename Lyt>
+struct random_layout_params
+{
+    /**
+     * Aspect ratio of the layout (i.e. the size of the layout).
+     */
+    typename Lyt::aspect_ratio max_coordinate;
+    /**
+     * Number of SiDBs that are placed on the layout.
+     */
+    uint64_t number_placed_sidbs = 0;
+    /**
+     * If positively charged SiDBs should be prevented, SiDBs are not placed closer than the minimal_spacing.
+     */
+    bool prevent_positive_charges = true;
+    /**
+     * If positively charged SiDBs should be prevented, SiDBs are not placed closer than this value (2 cells as
+     * Euclidean distance by default).
+     */
+    double minimal_spacing = 2;
+    /**
+     * Maximal number of attempts to generate the layout.
+     */
+    uint64_t maximal_attempts = 10E6;
+};
+
+namespace detail
+{
+template <typename Lyt>
+class generate_random_layout_impl
+{
+  public:
+    generate_random_layout_impl(const random_layout_params<Lyt>& params, std::ostream& s,
+                                std::vector<Lyt>& all_layouts) :
+
+            parameter{params},
+            os{s},
+            previous_layouts{all_layouts}
+    {}
+
+    void run()
+    {
+        bool successful_generation = false;
+        while (!successful_generation && parameter.maximal_attempts)
+        {
+            // layout is initialized with given aspect ratio and name.
+            Lyt lyt{parameter.max_coordinate};
+            // maximal x-coordinate is stored.
+            auto x_dimension = parameter.max_coordinate.x;
+            // maximal y-coordinate is stored.
+            auto                   y_dimension = parameter.max_coordinate.y;
+            static std::mt19937_64 generator(std::random_device{}());
+            uint64_t               attempt_counter = 0;
+            // this while stops if either all SiDBs are placed or the maximum number of attempts were performed.
+            while (lyt.num_cells() < parameter.number_placed_sidbs && attempt_counter < parameter.maximal_attempts)
+            {
+                // uniform distribution of [0,x_coordinate].
+                std::uniform_int_distribution<uint64_t> dist_x(0u, static_cast<uint64_t>(x_dimension));
+                // uniform distribution of [0,y_coordinate].
+                std::uniform_int_distribution<uint64_t> dist_y(0u, static_cast<uint64_t>(y_dimension));
+                // random integer from [0,x_coordinate] is selected.
+                const auto random_x_coordinate = dist_x(generator);
+                // random integer from [0,y_coordinate] is selected.
+                const auto random_y_coordinate = dist_y(generator);
+                // coordinate is constructed [x_coordinate, y_coordinate]
+                const auto random_coordinate = typename Lyt::coordinate({random_x_coordinate, random_y_coordinate});
```

Comment:
Actually, maybe a generic function to sample a random coordinate on any layout can be helpful for other tasks as well. Could you extract this as a `layout_utils.hpp` function?

Reply:
Good point! I will create this function.